library/src/main/java/com/fraggjkee/smsconfirmationview/SmsConfirmationView.kt

```diff
@@ -100,6 +112,55 @@ class SmsConfirmationView @JvmOverloads constructor(
         }
     }
 
+    // Used to inflate and return the popup paste menu
+    private fun initPasteMenu() : PopupMenu {
+        val clipboard = getSystemService(context, ClipboardManager::class.java)
+        val pasteMenu = PopupMenu(this.context, this).apply {
+            inflate(R.menu.popup_menu_paste)
+            setOnMenuItemClickListener { menuItem ->
+                when (menuItem.itemId) {
+                    R.id.menu_item_paste -> {
+                        // Examines the item on the clipboard. If getText() doesn't return null,
+                        // the clip item contains the text. Assumes that this application can only
+                        // handle one item at a time.
+                        val item = clipboard?.primaryClip?.getItemAt(0)
+
+                        // Get the clipboard item text.
+                        val pasteData = item?.text?.toString() ?: ""
```

Comment:
maybe we can let `appendPaste` to work with `CharSequence` and avoid `toString` conversion? I think `appendPaste ` can handle this type without any additional changes